options/persistent.go

```diff
@@ -0,0 +1,101 @@
+package options
+
+import (
+	"github.com/EventStore/EventStore-Client-Go/client/filtering"
+	"github.com/EventStore/EventStore-Client-Go/persistent"
+	"github.com/EventStore/EventStore-Client-Go/stream_position"
+)
+
+type PersistentStreamSubscriptionOptions struct {
+	settings persistent.SubscriptionSettings
+	position stream_position.StreamPosition
+}
+
+func PersistentStreamSubscriptionOptionsDefault() PersistentStreamSubscriptionOptions {
+	return PersistentStreamSubscriptionOptions{
+		settings: persistent.SubscriptionSettingsDefault(),
+		position: stream_position.End(),
+	}
+}
+
+func (opts PersistentStreamSubscriptionOptions) Settings(settings persistent.SubscriptionSettings) PersistentStreamSubscriptionOptions {
+	opts.settings = settings
+	return opts
+}
+
+func (opts PersistentStreamSubscriptionOptions) Position(position stream_position.StreamPosition) PersistentStreamSubscriptionOptions {
+	opts.position = position
+	return opts
+}
+
+func (opts PersistentStreamSubscriptionOptions) GetSettings() persistent.SubscriptionSettings {
+	return opts.settings
+}
+
+func (opts PersistentStreamSubscriptionOptions) GetPosition() stream_position.StreamPosition {
+	return opts.position
+}
+
+type PersistentAllSubscriptionOptions struct {
+	settings persistent.SubscriptionSettings
+	position stream_position.AllStreamPosition
+	filter   []filtering.SubscriptionFilterOptions
+}
+
+func PersistentAllSubscriptionOptionsDefault() PersistentAllSubscriptionOptions {
+	return PersistentAllSubscriptionOptions{
+		settings: persistent.SubscriptionSettingsDefault(),
+		position: stream_position.End(),
+		filter:   []filtering.SubscriptionFilterOptions{},
+	}
+}
+
+func (opts PersistentAllSubscriptionOptions) Settings(settings persistent.SubscriptionSettings) PersistentAllSubscriptionOptions {
```

Comment:
Not using a pointer here for the "self" parameter copies the `PersistentAllSubscriptionOptions` struct by value. I suggest changing these to use a pointer (ie `(opts * PersistentAllSubscriptionOptions)`) to be more efficient. The only caveat is it will be harder to call temporary values since those aren't addressable, ie `PersistentAllSubscriptionOptions{ settings: settings, position: position, filter: filter }.GetFilter()` won't work, since go can't take the address of a temporary value.